.gitignore

```diff
@@ -20,3 +20,4 @@ test/temp/
 .project
 .settings
 test/**/*.png
+dist
```

Comment:
we should keep dist at git.
It allows install from git and we won’t need to build at every test/integration test

Reply:
Are you sure we want them to be committed? How do we make sure they always represent the current state of the source files? e.g. someone forgets to commit the build files after doing some changes to the source files.

Reply:
I did the same thing in ionic-appauth so it's easy for developers to depend on the `main` branch if they want to be on the bleeding edge.

A precommit hook handles keeping the files up-to-date:

https://github.com/wi3land/ionic-appauth/blob/master/.husky/pre-commit

Reply:
@mraible not sure, but prepare script can be a problem at libs:
https://github.com/wi3land/ionic-appauth/blob/fa7574a7388b7a612c0efddb800ac95d15908a99/package.json#L11
I had some experience were it is called at a child project without support for husky breaking `npm install` at this project.
That’s the reason `generate-blueprint` command doesn’t generate husky by default:
https://github.com/mshima/generator-jhipster/blob/498979ef2ddbb46ec9a29078b5a5f4dff54cff98/generators/generate-blueprint/generator.mjs#L167

We can add tests to see if it’s updated at GitHub actions workflow.

Reply:
Some arguments I can think of against committing the build artifacts:
- there must be a system in place which makes sure the source files stay in sync with the build output
- the files will show in PR view and it will be harder to navigate
- VSCode will not ignore the build output in file searches by default

In my opinion, the downsides are pretty big so it'll like to explore some different solutions.

By installing from git you mean this, right? `npm install -g git://github.com:jhipster/generator-jhipster.git`

The blog post @mraible posted looks quite promising. Just running our `npm run build` script shouldn't fail like running `husky install` might. 

For optimizing the pipeline we could run the build and save the result (e.g. as a GitHub artifact) or publish the package (e.g. to GitHub packages).

Reply:
> By installing from git you mean this, right? `npm install -g git://github.com:jhipster/generator-jhipster.git`

Install locally like this:
https://github.com/hipster-labs/generator-jhipster-entity-audit/blob/4779ae2fb859d1c2ddc40ab2cd974dc65c06c6b2/package.json#L57

The blog post approach won't work because:
- sources are not included in package.json files.
- blueprint can have different typescript versions, they can be source incompatible.
- `generator-jhipster` project is quite big.

Since the `dist` directory will be too big with every templates (duplicated), we should not commit it to the repository.
Our release release process is quite complicated, using a snapshot can be help with development process.
We should think in something else later like publishing a build branch based on main.